Changes to reflect inverted classroom approach to JSL

shrutivanw opened the pull request:
Based on Cohort 8's JSL experience and student feedback, JSL will be following inverted classroom approach starting with Cohort 9.

Changes made:
- reflect the inverted classroom model we'll be following for JSL by leveraging Susan's excellent videos.
- clarify the criteria to be "students who are not yet well versed with loops, arrays and hashes" instead of "students who are not yet comfortable or confident in using loops, arrays and hashes".
- capitalize "S" in JumpStart for consistency in all our communication (welcome guide, other references in email etc.)

Changes not yet made: update JumpStart Live overview video to reflect inverted classroom approach.

shrutivanw committed:
Changes to reflect inverted classroom approach to JSL

Based on Cohort 8's JSL experience and student feedback, JSL will be following inverted classroom approach.
Changes made:
- reflect the inverted classroom model we'll be following for JSL by leveraging Susan's excellent videos.
- clarify the criteria to be "students who are not yet well versed with loops, arrays and hashes" instead of "students who are not yet comfortable or confident in using loops, arrays and hashes".
- capitalize "S" in JumpStart for consistency in all our communication (welcome guide, other references in email etc.)

susanev commented:
I looked at this super quickly, can review more this weekend. In the future, I would really love if we could have some communication before large amounts of changes are completed.

susanev reviewed (changes requested):
- Why the change from "comfortable" to "well-versed"? At a minimum please explain what you mean by "well-versed" .. I don't think this is super obvious
- Why are we switching up the numbering format?
- Please list out the actual prep work they need to do, instead of just linking elsewhere, I tried really hard to avoid making the curriculum a maze of links
- Please avoid language like "listed below" as it is not accessible to people using screen readers

shrutivanw committed:
Explicitly list homework (avoid links to prep work)

Explcitly listing all homework instead of linking to prep work for the next day. Getting rid of "listed below" per PR feedback.

shrutivanw commented:
@susanev I sent you an email to provide more context. It also answers some of reasons for the changes, which you were inquiring about. 
I've also update the PR to incorporate your suggestion to list all homework explicitly (avoid links to prep-work for the next day) and remove the usage of "listed below".

susanev commented:
If we keep the numbering system the same we can insert items later without having to fix all the numbering. I believe this is standard and what is used in all the other ada curriculum materials. Can you please verify that Ada as a whole has decided to change the style of numbering before making this change here.

susanev commented:
I am still very worried about using the language "well-versed" here it does not at all explain that you are choosing students based on whether they used certain constructs in the coding challenge. If that is the deciding factor you should be up front and say that exactly (I would like to discuss why this was changed in person at some point, as I am worried you are going to be missing students that need extra support). That being said, I don't think this new and specific selection criteria belongs in this curriculum. For now, I'd prefer if you just removed that whole section and use email/survey to communicate these details.

susanev commented:
With regards to this "inverted" approach you are describing, I need to take a much closer look this weekend and then I can provide more feedback.

shrutivanw commented:
Interesting wrt numbering system! I'll revert that change so that the numbers are consistent with the rest of Ada textbook curriculum.

shrutivanw commented:
AFAIK the JSL selection criteria was always primarily based on code submitted for phase 2 of admissions' process. I've been continuing the criteria that Crystal shared last time during C8 JSL selection process. So, if it was a change, it happened before April/May 2017. The criteria does get communicated in email to the students.
Sure, it makes sense to not mention all the detailed criteria in the curriculum. I'll remove the sections.

shrutivanw committed:
Remove selection criteria, change numbering to be "1."

Removes sections that talk about JSL selection criteria.
All numbering is kept consistent at "1." for easy additions in the future.

susanev commented:
When I taught JSL I was extremely thoughtful with students to determine who should attend which included a survey that used the language "confident" and "comfortable" along with reviewing their code from JS and reaching out to specific students that I was unsure about. I worry about the students who used these constructs in the code challenge but who are not comfortable and confident with their understanding, but maybe you are catching those students another way. Again, probably an in-person convo at some point, thanks for removing the section.

susanev commented on expectations.md:
```diff
@@ -1,15 +1,12 @@
 ## Prep Work
-* All participating students are expected to complete all prep work for each class. You can find information about the prep work in each days folder
-* We also expect students to come to class with relevant ideas, and questions related to the class topics
+* JumpStart Live follows the inverted classroom model, allowing for more discussion and time for questions in classroom. All participating students are expected to complete all prep work for each class. You can find information about the prep work in each day's folder.
```
I don't see a huge benefit of using this term here, they don't know what is was previously so there is really no frame of reference for the inverting. If you want to instead specifically call out the importance of completing the prep work thats fine. Maybe .. _"All participating students are expected to complete all prep work for each class. This will allow more time for discussion and questions during class. You can find information about the prep work in each day's folder."_

shrutivanw replied:
Makes sense. Done.

susanev commented on lessons/day1/readme.md:
```diff
@@ -1,8 +1,10 @@
 ## Prep work
-* Review your [Madlibs](https://github.com/Ada-Developers-Academy/jump-start/tree/master/lessons/grammar) solution and be prepared to talk with others about your code
+* Review your [Madlibs](https://github.com/Ada-Developers-Academy/jump-start/tree/master/lessons/grammar) solution and be prepared to talk with others about your code.
+* Watch [Day 1 Videos](https://adaacademy.hosted.panopto.com/Panopto/Pages/Sessions/List.aspx?folderID=66c9e5fc-8a11-4db2-b0fe-ebc516d45071) and review [lesson notes](#lessons).
```
Please make the review the lesson notes its own bullet point so students don't miss it; and for all following prep work items that have that as well

shrutivanw replied:
Makes sense. Done.

susanev commented on readme.md:
```diff
@@ -1,22 +1,24 @@
-# Jumpstart Live (JSL)
-_In person program for [Ada Developers Academy](http://adadevelopersacademy.org/) students who are not yet confident using loops, arrays, and hashes_
+# JumpStart Live (JSL)
+_In person program for [Ada Developers Academy](http://adadevelopersacademy.org/) students who are not yet well-versed using loops, arrays, and hashes_
```
just say .. for _some_ students or similar and drop the well-versed please

shrutivanw replied:
Done.

susanev commented on readme.md:
```diff
@@ -1,22 +1,24 @@
 
 ## Overview
-Launched in January 2016, to support newly admitted Ada Developers Academy students who are not yet comfortable and confident using loops, arrays, and hashes. This JSL curriculum is a companion to the [Jumpstart curriculum](https://github.com/Ada-Developers-Academy/jump-start).
+JumpStart Live (JSL) was launched in January 2016 to support newly admitted Ada Developers Academy students who are not yet well-versed using loops, arrays, and hashes. This JSL curriculum is a companion to the [JumpStart curriculum](https://github.com/Ada-Developers-Academy/jump-start).
```
drop _who are not yet well-versed using loops, arrays, and hashes_ please

shrutivanw replied:
Done.

susanev commented on readme.md:
```diff
@@ -1,22 +1,24 @@
 * General class plan
 	* Welcome, goals, warm-up (25 mins)
-	* Lesson (25 mins)
+	* Follow up lesson (25 mins)
```
Is this becoming "Follow up" lesson because its a follow up to their prep work? Or Follow up to what .. I think you can just keep this as Lesson unless you are trying to communicate something else here ..

shrutivanw replied:
I agree, it makes sense to just keep it as Lesson. Updated.

susanev commented on readme.md:
```diff
@@ -1,22 +1,24 @@
+	* Follow up lesson (25 mins)
 	* Break (10 mins)
-	* Exercises (50 mins)
+	* Exercise discussions (50 mins)
```
Will it only be discussion on the exercises? Or the other prep work too? You might consider just calling this section Discussion or something to be super clear and also value the other prep work they are doing.

shrutivanw replied:
Makes sense. Done.

susanev commented on remote-expectations.md:
```diff
@@ -1,23 +1,20 @@
 
 ## Work Submissions
-When you have completed all the work for a particular day, you should send a gist to the lead Jumpstart Instructor on slack. There are 6 total days, so this should be 6 total gists, but feel free to split the work up into more gists if that makes sense for you.
+When you have completed all the work for a particular day, you should send a gist to the lead JumpStart Instructor. There are 6 total days, so this should be 6 total gists, but feel free to split the work up into more gists if that makes sense for you.
```
Change to _at least 6 gists_ if you are allowing them to do more

shrutivanw replied:
Done.

susanev reviewed (changes requested):
A number of inline comments

susanev commented:
I believe this new approach is going to significantly increase the student time outside of the classroom (for the prep work+homework). Assuming this is something that Ada as agreed to require of students I would like to have this specifically called out in the class structure section. I am estimating times here based on what you added, but change if you have different thoughts on how much time it will take.

## Class details
### Time commitment
- 2 hours in person, twice a week for 3 weeks
- 6 hours of prep work before each class
- 2 hours of homework after each class

### In-person class structure
- ...

susanev commented:
Also curious about _"Share the gist with the instructor(s)."_ that you have added to the homework. Will you be giving each student feedback?

shrutivanw commented:
I added a section on Time Commitment. My estimates are different than yours. I'll stick with mine for C9 JSL and will ask for C9 feedback to improve the estimates.

Correct; the plan is that the instructors will be reviewing and providing feedback for each student for each day's homework.

shrutivanw committed:
Incorporating Susan's feedback

Made reviewing lesson notes a separate bullet point for prep work.  Added a section on time commitment. Removed selection criteria details.

shrutivanw committed:
Update homework section

Update homework section for each day to make reviewing lesson notes a separate bullet point (consitent with prep work).

shrutivanw committed:
Fit-and-finish: correcting a typo

Correcting a typo - adding a bullet point.

shrutivanw commented:
Thank you, Susan! PR updated to incorporate your feedback.

brookskindle commented:
I'm late to the code review party, but thank you @shrutivanw for keeping the repository up to date - and thank you @susanev for doing an awesome job reviewing it.
